block/manager_test.go

```diff
@@ -94,3 +95,88 @@ func TestIsDAIncluded(t *testing.T) {
 	m.blockCache.setDAIncluded(hash.String())
 	require.True(m.IsDAIncluded(hash))
 }
+
+func TestSubmitBlocks(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+
+	m := getManager(t)
+
+	maxDABlobSizeLimit, err := m.dalc.DA.MaxBlobSize(ctx)
+	require.NoError(err)
+
+	testCases := []struct {
+		name                        string
+		blocks                      []*types.Block
+		isErrExpected               bool
+		expectedPendingBlocksLength int
+	}{
+		{
+			name:                        "happy path, all blocks A, B, C are submitted on first round",
+			blocks:                      []*types.Block{types.GetRandomBlock(1, 5), types.GetRandomBlock(2, 5), types.GetRandomBlock(3, 5)},
+			isErrExpected:               false,
+			expectedPendingBlocksLength: 0,
+		},
+		{
+			name: "blocks A and B are submitted first round because including c triggers blob size limit. C is submitted on second round",
```

Comment:
```suggestion
			name: "blocks A and B are submitted together without C because including C triggers blob size limit. C is submitted in a separate round",
```